### PR TITLE
feat: migrate bundling to webpack 5

### LINKED ISSUE #70: Migrate UI bundling from rollup to webpack
since there is now an official svelte-loader for webpack, i could take a stab at migrating the bundling to webpack 5
the advantage is that there is only one bundler to worry about

### PR DESCRIPTION
**Pull request by burhanuday**
Migrated all bundling config to webpack 5
Deleted rollup config and unused packages

Closes #70

### CODE REVIEW

**Comment by burhanuday** [comment]
@bernaferrari bump

**Comment by bernaferrari** [comment]
Hey! Thanks a lot! I'll review really really soon.

**Comment by bernaferrari** [comment]
It doesn't work here :(

Try `yarn dev` then change something, like "Tailwind is a utility-first" to "Tailwind is a utility-second" and re-open the plugin in Figma (ctrl+alt+p), it should change the text.

**Comment by bernaferrari** [comment]
Hey, any news?

**Comment by burhanuday** [comment]
have figured it out!! The caveat is that HMR doesnt work but everything gets powered by a single webpack config and with caching, builds take less than a second. Will contribute by tomorrow

**Comment by burhanuday** [comment]
@bernaferrari hot reloading works for the code file but not for the ui code. this is because all ui code has to be inlined into the html file. but since that is only done once when the server starts and not on every change, it does not work for changes made after server has started

**Comment by burhanuday** [comment]
i might need to make a custom webpack plugin that takes the current one and also emits a new index.html whenever the ui plugin changes

**Comment by burhanuday** [comment]
@bernaferrari Fixed the issue without having to create a new webpack plugin. you can test it out now. first compile takes around 16 seconds, after that, new changes take 3-4 seconds to compile. i am on an intel mac so it should be faster on M1 or later

File: webpack.config.js [hunk 1]
```diff
@@ -1,58 +1,111 @@
...
+    filename: "[name].js",
+    path: path.resolve(__dirname, "dist"), // Compile into a folder called "dist"
+    clean: true,
+    publicPath: "/",
+  },
+  entry: {
+    ui: "./src/main.js", // The entry point for UI code
+    code: "./src/code.ts", // The entry point for plugin code
+  },
+  devtool: prod ? false : "inline-source-map",
```
Comment (burhanuday) [inline webpack.config.js:26]: you can make this false if you dont use this feature. i dont find it very helpful so leaving the choice upto you

File: webpack.config.js [hunk 2]
```diff
@@ -1,58 +1,111 @@
...
-        rules: [
-            {
-                test: /\.tsx?$/,
-                use: 'ts-loader',
-                exclude: /node_modules/,
+    extensions: [".tsx", ".ts", ".js", ".svelte"],
+    mainFields: ["svelte", "browser", "module", "main"],
+  },
+  optimization: {
+    minimize: true,
```
Comment (burhanuday) [inline webpack.config.js:35]: when minimize is false, svelte loader outputs some strange code which breaks the UI. probably experiment with this and raise an issue on that project

File: webpack.config.js [hunk 3]
```diff
@@ -1,58 +1,111 @@
...
+  },
+  module: {
+    rules: [
+      {
+        test: /\.tsx?$/,
+        use: [
+          {
+            loader: "ts-loader",
+            options: {
+              transpileOnly: true,
```
Comment (burhanuday) [inline webpack.config.js:46]: turned off type checking in the loader for a faster output. typechecking is anyway done by vscode. you can add https://www.npmjs.com/package/fork-ts-checker-webpack-plugin if you want to still do typechecking on a parallel thread that wont slow down typechecking

File: webpack.config.js [hunk 4]
```diff
@@ -1,58 +1,111 @@
...
+      {
+        test: /\.svelte$/,
+        use: {
+          loader: "svelte-loader",
+          options: {
+            compilerOptions: {
+              dev: false,
+            },
+            emitCss: true,
+            hotReload: false,
```
Comment (burhanuday) [inline webpack.config.js:61]: dev options turned off because we cant use dev server with figma plugins

File: webpack.config.js [hunk 5]
```diff
@@ -1,58 +1,111 @@
...
+      global: {}, // Fix missing symbol error when running in developer VM
+    }),
+    new HtmlWebpackPlugin({
+      inject: "body",
+      template: "./src/template.html",
+      filename: "index.html",
+      chunks: ["ui"],
+      cache: false,
+    }),
+    new InlineChunkHtmlPlugin(HtmlWebpackPlugin, [/ui/]),
```
Comment (burhanuday) [inline webpack.config.js:102]: this inlines the ui.js file into index.html since figma does not allow importing javascript from external files

**Comment by burhanuday** [comment]
@bernaferrari friendly ping

**Comment by bernaferrari** [comment]
I have a big update coming soon 👀

But yeah, I kind of forgot. But I've been working on something huge recently 😛.

**Comment by bernaferrari** [comment]
Plugin is now using vite :P but thanks for the help!

**Pull request closed without merging**